### PR TITLE
bws-68: + password validation worker

### PR DESCRIPTION
**Pull request by sofia0602**

### CODE REVIEW

File: shared/src/validation/eam-worker/worker-create/worker-create.validation-schema.ts [hunk 1]
```diff
@@ -18,6 +18,17 @@ const EamWorkerCreate = Joi.object({
       'string.min': EAMWorkerValidationMessage.NAME_MIN_LENGTH,
       'string.pattern.base': EAMWorkerValidationMessage.NAME_REGEX,
     }),
+  [getNameOf<EAMWorkerCreateRequestDto>('password')]: Joi.string()
```
Comment (what1s1ove) [inline shared/src/validation/eam-worker/worker-create/worker-create.validation-schema.ts]: we use this validation for the frontend. there is no `password` field. you need to create a separate one

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/validation-schemas/validation-schemas.ts [hunk 2]
```diff
@@ -1 +1,2 @@
 export { eamMasterSignUp, eamMasterSignIn } from './eam-master/eam-master';
+export { EamWorkerCreate } from './eam-worker/eam-worker';
```
Comment (Yaroslaww-1) [inline backend/src/validation-schemas/validation-schemas.ts]:
```suggestion
export { eamWorkerCreate } from './eam-worker/eam-worker';
```

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: shared/src/common/enums/validation/validation.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 export {
   EAMWorkerValidationMessage,
   EAMWorkerValidationRule,
+  EAMWorkerValidationMessageBackend,
```
Comment (what1s1ove) [inline shared/src/common/enums/validation/validation.ts]: you don't need to create new enums
you need to add messages and rules to `EAMWorkerValidationMessage`, `EAMWorkerValidationRule`

File: shared/src/common/enums/validation/worker/eam-worker-create-backend/eam-worker-create-validation-message.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { EAMWorkerValidationRuleBackend } from './eam-worker-create-validation-rule.enum';
+const EAMWorkerValidationMessageBackend = {
```
Comment (what1s1ove) [inline shared/src/common/enums/validation/worker/eam-worker-create-backend/eam-worker-create-validation-message.ts]: remove this file

File: shared/src/common/enums/validation/worker/eam-worker-create-backend/eam-worker-create-validation-rule.enum.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const EAMWorkerValidationRuleBackend = {
```
Comment (what1s1ove) [inline shared/src/common/enums/validation/worker/eam-worker-create-backend/eam-worker-create-validation-rule.enum.ts]: remove this file

File: shared/src/validation/eam-worker/eam-worker-create-backend/eam-worker-create.validation-schema.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import * as Joi from 'joi';
+import { getNameOf } from '~/helpers/helpers';
+import { EAMWorkerCreateRequestDto } from '~/common/types/types';
+import {
+  EAMWorkerValidationMessageBackend,
+  EAMWorkerValidationRuleBackend,
+} from '~/common/enums/enums';
+
+const eamWorkerCreate = Joi.object({
+  [getNameOf<EAMWorkerCreateRequestDto>('password')]: Joi.string()
```
Comment (what1s1ove) [inline shared/src/validation/eam-worker/eam-worker-create-backend/eam-worker-create.validation-schema.ts:21]: you also need to validate `name`, `groupsIds` and etc (check `EAMWorkerCreateRequestDto` and validate all fields)

File: shared/src/validation/eam-worker/worker.ts [hunk 7]
```diff
@@ -1 +1 @@
-export { EamWorkerCreate } from './worker-create/worker-create.validation-schema';
+export { eamWorkerCreate } from './eam-worker-create-backend/eam-worker-create.validation-schema';
```
Comment (what1s1ove) [inline shared/src/validation/eam-worker/worker.ts]: in this case, you break frontend validation 🤔
you need to keep both

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Pull request closed without merging**